cli/tests/test_immutable_commits.rs

```diff
@@ -109,6 +109,21 @@ fn test_rewrite_immutable_generic() {
     "###);
 }
 
+#[test]
+fn test_new_wc_commit_when_wc_immuable() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init"]);
+    test_env.jj_cmd_ok(test_env.env_root(), &["branch", "create", "main"]);
+    test_env.add_config(r#"revset-aliases."immutable_heads()" = "main""#);
+    test_env.jj_cmd_ok(test_env.env_root(), &["new", "-m=a"]);
+    let (_, stderr) = test_env.jj_cmd_ok(test_env.env_root(), &["branch", "set", "main"]);
+    insta::assert_snapshot!(stderr, @r###"
+Warning: The working-copy commit became immutable, so a new commit has been created on top of it.
+Working copy now at: zsuskuln 87e33403 (empty) (no description set)
+Parent commit      : kkmpptxz 7272528e main | (empty) a
+    "###);
+}
+
```

Comment:
It would be good to also have tests for these cases:

* Modify a file, change configuration to make the working copy immutable, then run e.g. `jj status`. This is the case we don't currently handle, but it would be good to still have coverage of the behavior even if it's not ideal
* Modify a file, set `immutable_heads()` to `@` and check that it doesn't run into some kind of infinite loop.

Reply:
What should be the expected behavior if @ is set immuatble? each transaction should create one new wc commit even if it's gonna be immutable too ? 

I could also check if the created commit if immutable, and if it's the case just discard it

Reply:
> What should be the expected behavior if `immutable_heads()` is set to @? each transaction should create one new wc commit even if it's gonna be immutable too ?

That seems reasonable and simplest, so that's what I would do.

Reply:
let's say user is on root commit.
user do `jj new`, then the newly created commit will be immutable as `immutable_heads()=@".
My current implementation will create another new commit on top of it (root -> empty immutable from new -> empty immutable from me). 

Shall i add an exception for that case ?

Reply:
I would not bother to do anything special for that case for now because configuring `immutable_heads=@` is such a weird thing to do. If we hear from users that there's a more reasonable way to run into a similar case, we can spend more effort on it then. But that's just my opinion.

Reply:
i agree with you, don't see the point of making `immutable_heads=@` in any case.